### DISCUSSION TITLE
How to merge with custom lua workspace library path?

### DISCUSSION THREAD

**Discussion by thenbe** [folke/lua-dev.nvim#152] [Q&A]
With the following config:

If I disable `neodev`, I get types for `awesome`, but not for `vim`.

If I enable `neodev`, I get types for `vim`, but not for `awesome`.

```lua
-- lua lsp config options
local options = {
  on_attach = on_attach,
  capabilities = capabilities,
  settings = {
    Lua = {
      workspace = {
        library = {
          ["/nix/store/x7wr7k6rl0jz4rpznp8fd0282gpvaccc-awesome-4.3/share/awesome/lib"] = true,
        },
      },
    },
  },
}

require("lspconfig").lua_ls.setup(options)
```

```jsonc
// .neoconf.json
{
  "neodev": {
    "library": {
      "enabled": true, // vim types work, awesome types don't work
      "enabled": false, // vim types don't work, awesome types work
      "plugins": true
    }
  },
  "neoconf": {
    "plugins": {
      "lua_ls": {
        "enabled": true
      }
    }
  }
}
```

It seems `neodev` is overriding the `Lua.workspace.library` key. What's a good approach to take here for enabling both at the same time? Thank you!

**Comment by mawkler** [comment]
I'm wondering the same thing :)

**Comment by mgastonportillo** [comment]
I have the same situation with nvchad and its types. With neodev enabled and properly configured, I get `vim.uv` types but I lose nvchad types. My temporary workaround was to disable neodev and manually add the types from `https://raw.githubusercontent.com/Bilal2453/luvit-meta/main/library/uv.lua` into my runtime path, but this is far from an optimal solution.
